src/pages/Apps/PlugConfigEdit.tsx

```diff
@@ -49,40 +53,34 @@ export function PlugConfigEdit({ slug }: Props) {
     }
   }, [existingConfig]);
 
+  const createOrUpdateConfig = useMutation({
+    mutationFn: isNew
+      ? mutate(routes.plugConfig.createPlugConfig)
+      : mutate(routes.plugConfig.updatePlugConfig, { pathParams: { slug } }),
+    onSuccess: () => navigate("/apps"),
+    onError: (error) => console.error("Error saving config:", error),
+  });
+
+  const deleteConfig = useMutation({
+    mutationFn: mutate(routes.plugConfig.deletePlugConfig, {
+      pathParams: { slug },
+    }),
+    onSuccess: () => navigate("/apps"),
+    onError: (error) => console.error("Error deleting config:", error),
+  });
+
   const handleSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
-    // Meta is parsed to a JSON object to clear whitespaces when saving
     const meta = JSON.parse(config.meta);
     const configPayload = { ...config, meta };
-    try {
-      if (isNew) {
-        await request(routes.plugConfig.createPlugConfig, {
-          body: configPayload,
-        });
-      } else {
-        await request(routes.plugConfig.updatePlugConfig, {
-          pathParams: { slug },
-          body: configPayload,
-        });
-      }
-      navigate("/apps");
-    } catch (error) {
-      console.error("Error saving config:", error);
-    }
+    createOrUpdateConfig.mutate(configPayload);
   };
 
-  const handleDelete = async () => {
-    try {
-      await request(routes.plugConfig.deletePlugConfig, {
-        pathParams: { slug },
-      });
-      navigate("/apps");
-    } catch (error) {
-      console.error("Error deleting config:", error);
-    }
+  const handleDelete = () => {
+    deleteConfig.mutate(undefined);
```

Comment:
ensure the routes schema's TBody is of type void, so that we don't need to pass undefined here.
```suggestion
    deleteConfig();
```